### PR TITLE
2.9.x scrubbing

### PR DESCRIPTION
**Pull request by JeffreyDevloo**

### CODE REVIEW

File: ovs/lib/helpers/generic/scrubber.py [hunk 1]
```diff
@@ -0,0 +1,1066 @@
...
+        :rtype: tuple(list, list)
+        :raises: ValueError: When an irregular item has been detected
+        """
+        if any(not isinstance(v, dict )for v in relevant_values):
+            raise ValueError('Not all relevant values are a dict')
+        if not isinstance(relevant_keys, list):
+            raise ValueError('The relevant keys should be a list of keys that are relevant')
+        if any(set(v.keys()) != set(relevant_keys) for v in relevant_values):
+            raise ValueError('The relevant values do not match the relevant format')
+        # Fetch all items, yield empty list if None
```
Comment (simonOpenV) [inline ovs/lib/helpers/generic/scrubber.py]: invalid comment

File: ovs/lib/helpers/generic/scrubber.py [hunk 2]
```diff
@@ -0,0 +1,1066 @@
...
+        fetched_items = self._fetch_registered_items()
+        relevant_work_items = []
+        # Filter out the relevant items
+        try:
+            for item in fetched_items or []:
+                # Extract relevant context. Note this is just a shallow copy and when retrieving items with lists/dicts we should not modify these in any way
+                # because the referance is kept in _relevant_work_items
```
Comment (simonOpenV) [inline ovs/lib/helpers/generic/scrubber.py]: typo in reference

File: ovs/lib/helpers/generic/scrubber.py [hunk 3]
```diff
@@ -0,0 +1,1066 @@
...
+                relevant_context = dict((k, v) for k, v in item.iteritems() if k in relevant_keys)
```
Comment (sjovdnbos) [inline ovs/lib/helpers/generic/scrubber.py]: typo in reference

File: ovs/lib/helpers/generic/scrubber.py (hunk 1 above)
Comment (sjovdnbos) [inline ovs/lib/helpers/generic/scrubber.py]: Invalid comment

File: ovs/lib/helpers/generic/scrubber.py [hunk 4]
```diff
@@ -0,0 +1,1066 @@
...
+        :param relevant_values: The values to check relevancy on (Only supporting dict types)
+        :type relevant_values: list[dict]
+        :param relevant_keys: The keys that are relevant for checking relevancy
+        (found items will strip keys to match to this format) (this format will be used to check in relevant values)
+        :type relevant_keys: list
+        :return: All relevant items and all fetched items
+        :rtype: tuple(list, list)
+        :raises: ValueError: When an irregular item has been detected
+        """
+        if any(not isinstance(v, dict )for v in relevant_values):
```
Comment (sjovdnbos) [inline ovs/lib/helpers/generic/scrubber.py]: whitespace in (v, dict_)

File: ovs/lib/helpers/generic/scrubber.py [hunk 5]
```diff
@@ -0,0 +1,1066 @@
...
+                max_tries = 5
+                while client is None:
+                    tries += 1
+                    if tries > max_tries:
+                        self._logger.error(self._format_message('Assuming StorageRouter {0} is dead. Not scrubbing there'.format(storagerouter.ip)))
+                        break
+                    try:
+                        # Requesting new client to avoid races (if the same worker would build the clients again)
+                        client = SSHClient(storagerouter, username='root', timeout=30, cached=False)
+                    except:
```
Comment (sjovdnbos) [inline ovs/lib/helpers/generic/scrubber.py]: bare except

File: ovs/lib/helpers/generic/scrubber.py [hunk 6]
```diff
@@ -0,0 +1,1066 @@
...
+                continue
+            if vd.is_vtemplate is True:
+                self._logger.info('{0} is a template, not scrubbing'.format(logging_start_vd))
+                continue
+            vd.invalidate_dynamics('storagedriver_id')
+            if not vd.storagedriver_id:
+                self._logger.warning('{0} no StorageDriver ID found'.format(logging_start_vd))
+                continue
+            work_queue.put(vd.guid)
+        total_items = list(itertools.chain(relevant_work_items, (self._wrap_data(item) for item in work_queue.queue)))
```
Comment (sjovdnbos) [inline ovs/lib/helpers/generic/scrubber.py]: Since this is a simple add operation, i'd prefer a '+' for the sake of comprehensibility

**Review by sjovdnbos — APPROVED** [review]

**Review by sjovdnbos — APPROVED** [review]

**Pull request merged**